libs/platform/ebpf_epoch.c

```diff
@@ -61,25 +70,31 @@ typedef struct _ebpf_epoch_state
     bool timer_disabled : 1; // Prevent re-arming the timer during shutdown.
 } ebpf_epoch_state_t;
 
+typedef struct _ebpf_epoch_thread_entry
+{
+    ebpf_epoch_state_t epoch_state;     // Per-thread epoch state.
+    uintptr_t old_thread_affinity_mask; // Thread affinity mask before entering an entry/exit block.
+    uintptr_t thread_id;                // Thread ID of the thread that owns this entry or 0 if unused.
+} ebpf_epoch_thread_entry_t;
+
+// The epoch code limits the number of threads that can be active at any one time per CPU.
+// If additional threads attempt to call ebpf_epoch_enter, they will block until a thread exits the epoch.
+#define EBPF_EPOCH_THREAD_ENTRY_TABLE_SIZE ((EBPF_CACHE_LINE_SIZE * 3) / sizeof(ebpf_epoch_thread_entry_t))
```

Comment:
why a multiple of 3? Is there an 'ideal' number of active threads per cpu that we're aiming for? Serious question: how was this number determined?

Reply:
The goal is to limit the number of cache lines per CPU. 4 seems reasonable, so 1 line for the main epoch state + 3 lines for the per-thread state.